src/nfa/nfa.cc

```diff
@@ -623,49 +623,34 @@ bool Mata::Nfa::are_state_disjoint(const Nfa& lhs, const Nfa& rhs)
     return true;
 } // are_disjoint }}}
 
-void Mata::Nfa::make_complete(
+bool Mata::Nfa::make_complete(
         Nfa&             aut,
         const Alphabet&  alphabet,
         State            sink_state)
 {
-    std::list<State> worklist(aut.initial.begin(),
-                              aut.initial.end());
-    std::unordered_set<State> processed(aut.initial.begin(),
-                                        aut.initial.end());
-
-    if (aut.size() <= sink_state) {
-        aut.add_state(sink_state);
-    }
-
-    worklist.push_back(sink_state);
-    processed.insert(sink_state);
-    while (!worklist.empty())
-    {
-        State state = *worklist.begin();
-        worklist.pop_front();
+    bool was_something_added = false;
 
+    auto num_of_states = aut.size();
+    for (State state = 0; state < num_of_states; ++state) {
         std::set<Symbol> used_symbols;
-        if (!aut.delta.empty())
-        {
-            for (const auto &symb_stateset: aut[state]) {
-                // TODO: Possibly fix insert.
-                used_symbols.insert(symb_stateset.symbol);
-
-                const StateSet &stateset = symb_stateset.targets;
-                for (const auto &tgt_state: stateset) {
-                    bool inserted;
-                    tie(std::ignore, inserted) = processed.insert(tgt_state);
-                    if (inserted) { worklist.push_back(tgt_state); }
-                }
-            }
+        for (auto const &tran : aut.delta[state]) {
+            used_symbols.insert(tran.symbol);
         }
-
         auto unused_symbols = alphabet.get_complement(used_symbols);
         for (Symbol symb : unused_symbols)
         {
             aut.delta.add(state, symb, sink_state);
+            was_something_added = true;
+        }
+    }
+
+    if (was_something_added && num_of_states <= sink_state) {
+        for (Symbol symbol : alphabet.get_alphabet_symbols()) {
```

Comment:
Isn't the alphabet already computed somewhere? Is it neccessary to call get_alphabet_symbols()?

Reply:
Alphabet is an abstract class that does not have any idea how symbols are stored, so I need to call `get_alphabet_symbols` to get the set of symbols of the alphabet.